compiler/lib/driver.ml

```diff
@@ -325,11 +325,25 @@ let output formatter ~standalone ~custom_header ?source_map () js =
   Js_output.program formatter ?source_map js;
   if times () then Format.eprintf "  write: %a@." Timer.print t
 
-let pack ~global { Linker.runtime_code = js; always_required_codes } =
+let pack ~global ~standalone { Linker.runtime_code = js; always_required_codes } =
   let module J = Javascript in
   let t = Timer.make () in
   if times () then Format.eprintf "Start Optimizing js...@.";
   (* pre pack optim *)
+  let js =
+    let o = new Js_traverse.free in
+    let js = o#program js in
+    if StringSet.mem Constant.old_global_object o#get_free_name
+    then
+      ( J.Statement
+          (J.Variable_statement
+             [ ( J.ident Constant.old_global_object
```

Comment:
Not sure i'm reading this right, does this mean that existing usages of `joo_global_object` will compile down to `globalThis`?

I am thinking about cases where code would break after updating the version, due to the environment not supporting `globalThis` for some reason. If these assumptions are correct, would this patch be released under major version then?

Reply:
when emitting the js, if there are any references to "joo_global_object", we add a `var joo_global_object = globalThis`.
The code is then wrapped with
```js
(function(globalThis){
  var joo_global_object = globalThis;
  rest_of_the_code
})
(globalThis)`